### PR TITLE
Run migrations as part of deployment

### LINKED ISSUE #101: Task: Automatically run DB migrations
Deployment step should run migrations so we don't forget to do it manually

This task is scoped to:
- [ ] Run migrations as part of the deploy
- [ ] Determine if the deployed app needs to be restarted post-migration

### PR DESCRIPTION
**Pull request by rahearn**
## Description of change

Updates the deployment action to run migrations if there have been any changes in `db/migrate`

Also updated the rest of the actions to utilize the composite run actions as [nested `uses` is now supported](https://github.com/actions/runner/issues/646#issuecomment-901336347).

## Acceptance Criteria

When a deployment happens that involves a new migration, that migration is automatically run

## How to test

Tested locally, but this shouldn't do anything until we have our first story that involves the DB

## Issue(s)

* closes #101 

## Checklist

To be completed by the submitter:

- [x] Meets issue criteria
- [x] Project board status updated

## To the Reviewer

This project is using [Conventional Comments](https://conventionalcomments.org/) to give structure
and context to PR comments. Please use these labels in your comments.

* **praise:**
* **nitpick:**
* **suggestion:**
* **issue:**
* **question:**
* **thought:**
* **chore:**

### CODE REVIEW

File: .github/workflows/deploy-stage.yml [hunk 1]
```diff
@@ -26,3 +28,36 @@ jobs:
...
+        with:
+          cf_username: ${{ secrets.CF_USERNAME }}
+          cf_password: ${{ secrets.CF_PASSWORD }}
+          cf_org: hhs-acf-ohs-hses
+          cf_space: ct-stage
+          full_command: |
+            cf run-task complaints-tracker-stage -c "bundle exec rake db:migrate"
+
+
+      - name: "wait a minute for task to run"
```
Comment (rahearn) [inline .github/workflows/deploy-stage.yml:50]: Not an ideal solution, but `run-task` kicks off in the background and I don't think there's a good way to programmatically check for status updates.

File: .github/actions/setup-javascript/action.yml [hunk 2]
```diff
@@ -12,4 +10,5 @@ runs:
         cache: 'yarn'
 
     - name: Install yarn dependencies
+      shell: bash
```
Comment (jduss4) [inline .github/actions/setup-javascript/action.yml:13]: [question] Do you know when specifying bash is necessary?

**Review by jduss4 — APPROVED** [review]

File: .github/actions/setup-javascript/action.yml (hunk 2 above)
Comment (rahearn) [reply .github/actions/setup-javascript/action.yml:13]: You'd have to ask github. For whatever reason `run` steps default to bash when used in a workflow, but it's required to specify the shell when used in a `composite run steps` shared action.

Comment (rahearn) [reply .github/actions/setup-javascript/action.yml:13]: :sigh: just realized you said `when` and not `why`. The when is: anytime you're creating a shared action and anytime you want to use something other than bash.

Comment (jduss4) [reply .github/actions/setup-javascript/action.yml:13]: Ha, you actually answered my question with your first go! The docs I looked at were basically like "it uses bash by default except on Windows" and so I didn't really know why you were specifying it.  But, the composite run steps seems to be the answer!

File: terraform/README.md [hunk 3]
```diff
@@ -33,9 +33,11 @@ The below steps rely on you first configuring access to the Terraform state in s
     ```bash
     terraform init
     terraform plan
-    terraform apply
     ```
 
+1. GitHub Actions will apply the changes when your PR is merged. To force `terraform apply` to run even if you didn't
+update any terraform files, make any change to `terraform/<env>/.force-action-apply`
```
Comment (rahearn) [inline terraform/README.md:39]: Added this bit because `terraform apply` didn't run after the only change you had to make was updating the workflow file.

**Review by jduss4 — APPROVED** [review]

**Pull request merged**